src/metabase/models/user.clj

```diff
@@ -165,6 +184,32 @@
         (assoc user :has_invited_second_user (and (= (:id user) 1)
                                                   (> user-count 1)))))))
 
+(defn add-is-installer
+  "Adds the `is_installer` flag to a collection of `users`. This should be `true` for only the user who
+  underwent the initial app setup flow (with an ID of 1). This is used to modify the experience of the
+  starting page for users."
+  {:batched-hydrate :is_installer}
+  [users]
+  (when (seq users)
+    (for [user users]
+      (assoc user :is_installer (= (:id user) 1)))))
+
+(defn add-has-question-and-dashboard
+  "Adds the `has-question-and-dashboard` flag to a collection of `users`. This should be true for the user
+  if they have access to at least one question and one dashboard."
+  {:batched-hydrate :has_question_and_dashboard}
+  [users]
+  (let [perms-query (fn [user]
+                      {:where
+                       [:and
+                        [:= :archived false]
+                        (collection/visible-collection-ids->honeysql-filter-clause
+                         :collection_id
+                         (collection/permissions-set->visible-collection-ids (permissions-set user)))]})]
+    (when (seq users)
+      (for [user users]
+        (assoc user :has_question_and_dashboard (and (> (db/count 'Card (perms-query user)) 0)
```

Comment:
Two thoughts:
- `perms-query` makes two db calls each time and you invoke it twice instead of just once and reusing it
- `perms-query` makes two calls and we do it for each user. This could get really expensive depending on the number of users. Is it possible to batch this up and get some map of `user-id->permissions`?

Reply:
Oh, that's a good point. I'll rework this, thanks